### ISSUE TITLE
MD037 crashes for <li> with underscores

### ISSUE THREAD

**Issue by akoeplinger** [markdownlint/markdownlint#83]
This html in a markdown doc:

```
<ul>
    <li>5.4 Details for _write, _read operations</li>
</ul>
```

results in a crash:

```
[..]
Processing rule MD037
/home/alexander/dev/markdownlint/lib/mdl/doc.rb:222:in `block (2 levels) in matching_text_element_lines': undefined method `+' for nil:NilClass (NoMethodError)
    from /home/alexander/dev/markdownlint/lib/mdl/doc.rb:221:in `each'
    from /home/alexander/dev/markdownlint/lib/mdl/doc.rb:221:in `each_with_index'
    from /home/alexander/dev/markdownlint/lib/mdl/doc.rb:221:in `block in matching_text_element_lines'
    from /home/alexander/dev/markdownlint/lib/mdl/doc.rb:218:in `each'
    from /home/alexander/dev/markdownlint/lib/mdl/doc.rb:218:in `matching_text_element_lines'
    from /home/alexander/dev/markdownlint/lib/mdl/rules.rb:500:in `block (2 levels) in load'
    from /home/alexander/dev/markdownlint/lib/mdl.rb:71:in `call'
    from /home/alexander/dev/markdownlint/lib/mdl.rb:71:in `block (2 levels) in run'
    from /home/alexander/dev/markdownlint/lib/mdl.rb:69:in `each'
    from /home/alexander/dev/markdownlint/lib/mdl.rb:69:in `block in run'
    from /home/alexander/dev/markdownlint/lib/mdl.rb:59:in `each'
    from /home/alexander/dev/markdownlint/lib/mdl.rb:59:in `run'
    from /home/alexander/dev/markdownlint/bin/mdl:10:in `<top (required)>'
    from /home/alexander/.rvm/gems/ruby-2.1.2/bin/mdl:23:in `load'
    from /home/alexander/.rvm/gems/ruby-2.1.2/bin/mdl:23:in `<main>'
    from /home/alexander/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `eval'
    from /home/alexander/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `<main>'

```

**Comment by mivok** [comment]
This appears to be related to kramdown not providing line number information for the `<li>` tag when it's parsed, and consequently giving the text inside a location of 'nil' and throwing the error you see. This will affect any of MD011, MD034, MD037 if they match a line inside `<li>` elements.

However, even if this wasn't the case, your example line shouldn't have been detected as emphasis (it's just two words on a line beginning with an underscore), and so I've pushed a change that fixes this. This should fix your immediate problem while I look into the underlying issue.

**Comment by akoeplinger** [comment]
Thanks :+1: (and yes I saw the same crash on MD034 as well)

**Comment by mivok** [comment]
For now I've released v0.2.1 with fixes for the other issues reported, and will work on the kramdown issue for the next release. In the meantime, if this is a significant issue, I'd just recommend temporarily disabling MD011, MD034 and MD037 as a workaround.

**Comment by akoeplinger** [comment]
It's not a significant issue, I just disabled the rules as you said ;)

**Comment by mivok** [comment]
Opened https://github.com/gettalong/kramdown/issues/243 upstream to deal with the underlying issue.

**Comment by mivok** [comment]
This was fixed with kramdown 1.7.0 and above.

**Comment by akoeplinger** [comment]
@mivok Thanks, upgrading kramdown indeed fixed this particular crash :star2:. However after re-enabling MD034, I'm seeing a similar crash: https://github.com/mivok/markdownlint/issues/97
